reposilite-plugins/javadoc-plugin/src/main/kotlin/com/reposilite/plugin/javadoc/JavadocPlugin.kt

```diff
@@ -0,0 +1,39 @@
+package com.reposilite.plugin.javadoc
+
+import com.reposilite.Reposilite
+import com.reposilite.maven.MavenFacade
+import com.reposilite.plugin.api.Facade
+import com.reposilite.plugin.api.Plugin
+import com.reposilite.plugin.api.ReposiliteInitializeEvent
+import com.reposilite.plugin.api.ReposilitePlugin
+import com.reposilite.web.api.RoutingSetupEvent
+import java.io.File
+
+@Plugin(name = "JavadocPlugin")
+class JavadocPlugin : ReposilitePlugin() {
+
+    private lateinit var reposilite: Reposilite
+    private lateinit var javadocFolder: File
+
+    override fun initialize(): Facade? {
+        extensions().registerEvent(ReposiliteInitializeEvent::class.java) { event: ReposiliteInitializeEvent ->
```

Comment:
You can listen for events directly through `ReposilitePlugin#event` method, e.g.:

```kotlin
event { event: RoutingSetupEvent ->
    event.registerRoutes(MavenEndpoints(mavenFacade, frontendFacade, settingsFacade))
    event.registerRoutes(MavenApiEndpoints(mavenFacade))
    event.registerRoutes(MavenLatestApiEndpoints(mavenFacade, settingsFacade))
}
```